sdk/core/System.ClientModel/src/Pipeline/ClientPipeline.cs

```diff
@@ -144,6 +154,18 @@ public static ClientPipeline Create(ClientPipelineOptions? options = default)
 
         int perTryIndex = index;
 
+        if (isLoggingEnabled)
+        {
+            var loggingPolicy = options.LoggingPolicy ?? new ClientLoggingPolicy(options: diagnosticsOptions);
+            if (loggingPolicy is ClientLoggingPolicy clientLoggingPolicy)
+            {
+                var loggedHeaders = diagnosticsOptions?.LoggedHeaderNames?.ToArray() ?? Array.Empty<string>();
+                var loggedQueries = diagnosticsOptions?.LoggedQueryParameters?.ToArray() ?? Array.Empty<string>();
+                clientLoggingPolicy.Sanitizer = new PipelineMessageSanitizer(loggedQueries, loggedHeaders);
```

Comment:
Could this be initialized internally in the policy constructor and we remove the internal property?

There's also the question of whether we would want to reuse the sanitizer for a future distributed tracing policy and want to make it an internal property on the pipeline itself so multiple policies have access to it?

Reply:
Updated this. I think since this is internal we can consider moving it to the pipeline when the distributed tracing policy is added